src/allocator.c

```diff
@@ -90,10 +90,10 @@ rcutils_allocator_is_valid(const rcutils_allocator_t * allocator)
 void *
 rcutils_reallocf(void * pointer, size_t size, rcutils_allocator_t * allocator)
 {
-  if (!allocator || !allocator->reallocate || !allocator->deallocate) {
+  if (!rcutils_allocator_is_valid(allocator)) {
```

Comment:
Should this have the `#if RCUTILS_REPORT_ERROR_HANDLING_ERRORS` or equivalent?

Reply:
added in https://github.com/ros2/rcutils/pull/67/commits/b9c943a04e1b194be09e36513cbd85a3080a0428, I reused the error handling definition but am fine defining a new one if it's preferred